src/work.c

```diff
@@ -104,14 +107,30 @@ static void luv_work_cb(uv_work_t* req) {
     }
     luv_thread_arg_clear(L, &work->args, LUVF_THREAD_SIDE_CHILD);
   } else {
-    fprintf(stderr, "Uncaught Error: %s can't be work entry\n",
-            lua_typename(L, lua_type(L,-1)));
     lua_pop(L, 1);
     luv_thread_arg_clear(L, &work->args, LUVF_THREAD_SIDE_CHILD);
+    return luaL_error(L, "Uncaught Error: %s can't be work entry\n",
+            lua_typename(L, lua_type(L,-1)));
   }
   work->args.L = L;
   if (top!=lua_gettop(L))
-    luaL_error(L, "stack not balance in luv_work_cb, need %d but %d", top, lua_gettop(L));
+    return luaL_error(L, "stack not balance in luv_work_cb, need %d but %d", top, lua_gettop(L));
+  return LUA_OK;
+}
+
+static void luv_work_cb_wrapper(uv_work_t* req) {
+  luv_work_t* work =  (luv_work_t*)req->data;
+  luv_work_ctx_t* ctx = work->ctx;
+  lua_State *L = work->args.L;
+  luv_ctx_t* lctx = luv_context(L);
+
+  // If exit is called on a thread in the thread pool, abort is called in
+  // uv__threadpool_cleanup, so exit is not called in luv_cfpcall.
+  int i = lctx->thrd_cpcall(L, luv_work_cb, (void*)req, LUVF_CALLBACK_NOEXIT);
+  if (i != LUA_OK) {
```

Comment:
Lua vm must be push back to vm pools cache in main threads to fix https://github.com/luvit/luv/pull/566/checks?check_run_id=3639646912#step:4:9914

Maybe we can copy and refact code from 
https://github.com/luvit/luv/blob/b2572ecf988185b9763965c590f6edbb2bc9b6c3/src/work.c#L129-L138

Run here means not call luv_after_work_cb

Reply:
Or do a wrapper for luv_work_cb, insure luv_after_work_cb called everytime is a good idea.